cdap-hbase-compat-base/src/main/java/co/cask/cdap/replication/TableUpdater.java

```diff
@@ -70,9 +68,11 @@ public TableUpdater(String rowType, final Configuration conf) {
    * @param regionName is the key for the Map
    * @param time is the value for the Map
    */
-  public synchronized void updateTime(String regionName, long time) throws IOException {
-    if (!cachedUpdates.containsKey(regionName) || time > cachedUpdates.get(regionName)) {
-      cachedUpdates.put(regionName, time);
+  public void updateTime(String regionName, long time) throws IOException {
+    synchronized (cachedUpdates) {
```

Comment:
I'm confused again about the concurrency of the coprocessor. There can be concurrent postReplicate/postWALWrite calls happening on a single coprocessor? It may make sense for postReplication, but seems odd for postWALWrite, as I don't imagine HBase writes to the WAL in a concurrent way. 

If so, it means the previous logic could have ended up putting the wrong time in the map.  Also if so, this would essentially remove all concurrency. Do we know the performance impact?